File: alexis.rb
class Alexis

  attr_accessor :alexis, :swag

  def initialize
    @name = "alexis"
    @swag = 1.0/0.0
    self.proclaim_swagness
  end

  def proclaim_swagness
    puts("swag on #{@swag}")
  end
end

alexis = Alexis.new

File: alexis.cpp
#include <iostream>

using std::string;
using std::cout;
using std::endl; 

class Alexis {
  public:
    string name;
    string swag;
    Alexis() {
      name = "alexis";
      swag = "infinity";
      proclaimSwagness();
    }
    void proclaimSwagness() {
      cout << "swag on " << swag << endl;
    }
}

int main() {
  Alexis alexis;
}